File: apps/party-time-backend/src/main/java/com/partytime/api/dto/changepassword/ChangePasswordDTO.java
package com.partytime.api.dto.changepassword;

import com.partytime.api.ApiConstants;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class ChangePasswordDTO {

    @NotNull
    @NotEmpty
    @Size(min = 8, max = 30)
    @Pattern(regexp = ApiConstants.REGEX_PASSWORD)
    private String oldPassword;

    @NotNull
    @NotEmpty
    @Size(min = 8, max = 30)
    @Pattern(regexp = ApiConstants.REGEX_PASSWORD)
    private String newPassword;

}

File: apps/party-time-backend-kotlin/src/main/kotlin/com/partytime/api/dto/changepassword/ChangePasswordDTO.kt
package com.partytime.api.dto.changepassword

import com.partytime.api.ApiConstants
import jakarta.validation.constraints.NotEmpty
import jakarta.validation.constraints.Pattern
import jakarta.validation.constraints.Size

/**
 * Serializable information container for Password changes
 *
 * @param oldPassword The old password of the user whose password should be changed
 * @param newPassword The new password of the user whose password should be changed
 */
data class ChangePasswordDTO(
    @field:NotEmpty
    @field:Size(min = 8, max = 30)
    @field:Pattern(regexp = ApiConstants.REGEX_PASSWORD)
    val oldPassword: String,
    @field:NotEmpty
    @field:Size(min = 8, max = 30)
    @field:Pattern(regexp = ApiConstants.REGEX_PASSWORD)
    val newPassword: String
)
